src/newton1d.jl

```diff
@@ -28,56 +30,111 @@ function newton1d{T}(f::Function, f′::Function, x::Interval{T};
-                    push!(R, Root(X, :unique))
-
-                    debug && @show "Root found", X
+                elseif 0 ∈ f(interval(prevfloat(mid(X)), nextfloat(mid(X))))
+                    n = fa = fb = 0
+                    root_exist = true
+                    while (n < 4 && (fa == 0 || fb == 0))
+                        if fa == 0
+                            if 0 ∈ f(interval(prevfloat(X.lo), nextfloat(X.lo)))
+                                fa = 1
+                            else
+                                N = X.lo - (f(interval(X.lo)) / f′(X))
+                                X = X ∩ N
+                                if (isempty(X) || diam(X) == 0)
+                                    root_exist = false
+                                    break
+                                end
+                            end
+                        end
+                        if fb == 0
+                            if 0 ∈ f(interval(prevfloat(X.hi), nextfloat(X.hi)))
+                                fb = 1
+                            else
+                                if 0 ∈ f(interval(prevfloat(mid(X)), nextfloat(mid(X))))
+                                    N = X.hi - (f(interval(X.hi)) / f′(X))
+                                else
+                                    N = mid(X) - (f(interval(mid(X))) / f′(X))
+                                end
+                                X = X ∩ N
+                                if (isempty(X) || diam(X) == 0)
+                                    root_exist = false
+                                    break
+                                end
+                            end
+                        end
+                        N = mid(X) - (f(interval(mid(X))) / f′(X))
+                        X = X ∩ N
+                        if (isempty(X) || diam(X) == 0)
+                            root_exist = false
+                            break
+                        end
+                        n += 1
+                    end
+                    if root_exist
+                        push!(R, Root(X, :unique))
+                        debugroot && @show "Root found", X
+                    end
 
                     break
                 end
             end
 
         else
-            # 0 ∈ f'(X)
-
+            if diam(X) == initial_width
+                reps += 1
+                if reps > 10
+                    push!(R, Root(X, :unknown))
+                    debugroot && @show "Repititive root found", X
+                    reps = 0
+                    continue
+                end
+            end
+            initial_width = diam(X)
             debug && println("0 ∈ f'(X)")
 
             expansion_pt = Inf
             # expansion point for the newton step might be m, X.lo or X.hi according to some conditions
 
-            if 0 ∈ f(Interval(mid(X)))
+            if 0 ∈ f(interval(prevfloat(mid(X)), nextfloat(mid(X))))
```

Comment:
There is a function `IntervalArithmetic.wideinterval` for this. You could give it a shorter name or import it.

Reply:
But I'm not sure why you're doing this widening here?

Reply:
Okay, I will import `wideinterval`.
Using the normal interval was causing the method to getting stuck sometimes earlier, hence the widening. It further narrows the interval once `0 ∈ f(interval(prevfloat(mid(X)), nextfloat(mid(X))))` returns `true`.